mimic/test/dummy.py

```diff
@@ -195,7 +203,17 @@ def make_example_external_api(case, name=u"example",
     if endpoint_templates is None:
         endpoint_templates = [ExampleEndpointTemplate()]
 
-    service_type = endpoint_templates[0].type_key
+    if service_type is None:
+        # default parameter value, take the template type from the first
+        # endpoint template in the list
+        service_type = endpoint_templates[0].type_key
+
+    elif len(endpoint_templates) == 1:
+        # service_type was specified and most likely endpoint_templates
```

Comment:
The `most likely` wording is simply due to the fact that in most cases the default will be used. Still, after reviewing the function I think I've got a slight improvement that can remove the branch that wording falls under without having to add any additional variables to track the state.

The intent was not to remember anything, but to capture default values and ensure the initial template met the conditions of the docstring specification.